src/main/java/org/jenkinsci/test/acceptance/po/Control.java

```diff
@@ -94,10 +95,39 @@ public void check(boolean state) {
         check(resolve(), state);
     }
 
+    /**
+     * sends a click on the underlying element.
+     * You should not use this on any page where the click will cause another page to be loaded as it will not
+     * gaurantee that the new page has been loaded.
+     * @see #clickAndWaitToBecomeStale()
+     * @see #clickAndWaitToBecomeStale(Duration)
+     */
     public void click() {
         resolve().click();
     }
 
+
+    /**
+     * like click but will block for up to 30 seconds until the underlying web element has become stale.
+     * see https://blog.codeship.com/get-selenium-to-wait-for-page-load/
+     */
+    public void clickAndWaitToBecomeStale() {
+        clickAndWaitToBecomeStale(Duration.ofSeconds(30));
+    }
+
+    /**
+     * like click but will block until the underlying web element has become stale.
+     * see https://blog.codeship.com/get-selenium-to-wait-for-page-load/
+     * @param timeout the amount of time to wait
+     */
+    public void clickAndWaitToBecomeStale(Duration timeout) {
+        WebElement webElement = resolve();
+        // webElement.submit() despite advertising it does exactly this just blows up :(
+        webElement.submit();
```

Comment:
I don't get what you mean here - you say it blows up but yet it is still used?

Reply:
erm it shouldn't be used it was blowing up locally.  looks like I forgot to push some changes :-(

Reply:
Testing locally, `submit()` appears to work for me.  I am still reading through the selenium implementation to see if it might be susceptible to the same issues, but in any case i think having `clickAndWaitToBecomeStale(...)` (or possibly an overloaded version of `click()`) still makes sense to add as an API because the default behavior in non-intuitive and indeterminately used in our page objects.